src/main/java/com/jfrog/ide/idea/inspections/AbstractInspection.java

```diff
@@ -168,207 +178,61 @@ boolean isShowInspection(PsiElement element) {
-     */
-    DependencyTree getProjectNode(DependencyTree root, Project project) {
-        if (root.getGeneralInfo() != null) {
-            return root;
-        }
-        return root.getChildren().stream()
-                .filter(child -> StringUtils.equals((String) child.getUserObject(), project.getName()))
-                .findAny()
-                .orElse(root);
-    }
-
-    /**
-     * Get the root of the dependency tree of the input project.
-     *
-     * @param element - The Psi element in the package descriptor
-     * @return root of the dependency tree of the input project
-     */
-    DependencyTree getRootDependencyTree(PsiElement element) {
-        Project project = element.getProject();
-        ComponentsTree componentsTree = LocalComponentsTree.getInstance(project);
-        if (componentsTree == null || componentsTree.getModel() == null) {
-            return null;
-        }
-        return (DependencyTree) componentsTree.getModel().getRoot();
-    }
-
-    /**
-     * Return true if and only if the dependency tree node containing the input path.
-     *
-     * @param node - The dependency tree node
-     * @param path - The path to check
-     * @return true if and only if the dependency tree node containing the input path
-     */
-    private boolean isContainingPath(DependencyTree node, String path) {
-        GeneralInfo childGeneralInfo = node.getGeneralInfo();
-        return StringUtils.equals(childGeneralInfo.getPath(), path);
-    }
-
-    /**
-     * Get all submodules from the dependencies-tree which are containing the dependency element.
-     * Currently in use for Go and npm.
-     *
-     * @param root    - The root of the dependency tree
-     * @param element - The Psi element in the package descriptor
-     * @return Set of modules containing the dependency or null if not found
-     */
-    Set<DependencyTree> collectModules(DependencyTree root, PsiElement element) {
-        // Single project, single module
-        if (root.getGeneralInfo() != null) {
-            return Sets.newHashSet(root);
-        }
-
-        // Multi project
-        String path = element.getContainingFile().getVirtualFile().getParent().getPath();
-        for (DependencyTree child : root.getChildren()) {
-            if (isContainingPath(child, path)) {
-                return Sets.newHashSet(child);
-            }
-        }
-        return null;
-    }
-
-    /**
-     * Get all submodules from the dependencies-tree which are containing the dependency element.
-     * Currently in use for Gradle and Maven.
+     * Get the dependencies that match to the input componentName.
      *
-     * @param root          - The root of the dependency tree
-     * @param project       - The project
-     * @param modulesList   - List of all relevant modules
+     * @param file          - The Descriptor file node in the tree
      * @param componentName - Component name representing a dependency without version
```

Comment:
Why actually don't we compare also the component version? In most cases, we can get it from the PsiElement.